TombEngine/Game/control/box.cpp

```diff
@@ -137,77 +138,68 @@ bool MoveCreature3DPos(Pose* fromPose, Pose* toPose, int velocity, short angleDi
 	return false;
 }
 
-void CreatureYRot2(Pose* fromPose, short angle, short angleAdd) 
+void CreatureYRot2(Pose* fromPose, short angle, short angleAdd)
 {
-	if (angleAdd < angle)
+	if (angle > angleAdd)
 	{
 		fromPose->Orientation.y += angleAdd;
-		return;
-	} 
-
-	if (angle < -angleAdd)
+	}
+	else if (angle < -angleAdd)
 	{
 		fromPose->Orientation.y -= angleAdd;
-		return;
-	} 
-
-	fromPose->Orientation.y += angle;
+	}
+	else
+	{
+		fromPose->Orientation.y += angle;
+	}
 }
 
 bool SameZone(CreatureInfo* creature, ItemInfo* target)
 {
-	int* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
-	auto* item = &g_Level.Items[creature->ItemNumber];
-
-	auto* room = &g_Level.Rooms[item->RoomNumber];
-	FloorInfo* floor = GetSector(room, item->Pose.Position.x - room->x, item->Pose.Position.z - room->z);
-	if (floor->Box == NO_BOX)
-		return false;
-
-	item->BoxNumber = floor->Box;
```

Comment:
Why checks for NO_BOX were removed from this function? We still can face the situation when entity is pushed to a block without box (e.g. by vehicle or rollingball).